spec/HttpClient/HttpExceptionHandlerSpec.php

```diff
@@ -52,6 +53,23 @@ function it_throws_bad_request_exception_when_status_code_401(
             ->during('transformResponseToException', [$request, $response]);
     }
 
+    function it_throws_not_found_exception_when_status_code_404(
```

Comment:
Could you change the name of the other specs ? `it_throws_bad_request_exception_when_status_code_401` should be `it_throws_unauthorized_request_exception_when_status_code_401`. I did some copy/past I suppose sorry :/